File: app/src/main/java/com/brankas/testapp/fragment/ClientDetailsFragment.java
package com.brankas.testapp.fragment;

import android.os.Bundle;

import com.brankas.testapp.R;
import com.brankas.testapp.customview.BoxedEditText;
import com.brankas.testapp.listener.ScreenListener;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

public class ClientDetailsFragment extends BaseFragment {

   public static final String DISPLAY_NAME = "display_name";
   public static final String RETURN_URL = "return_url";
   public static final String FAIL_URL = "fail_url";
   public static final String EXTERNAL_ID = "external_id";

    @Override
    public int getLayoutId() {
        return R.layout.fragment_client_details;
    }

    @Override
    public void initDetails() { }

    @Override
    public void showError(String tag) {
        if(tag != null) {
            switch (tag) {
                case RETURN_URL:
                    ((BoxedEditText)parentView.findViewById(R.id.returnUrl))
                            .showError(getString(R.string.invalid_url_format));
                    break;
                case FAIL_URL:
                    ((BoxedEditText)parentView.findViewById(R.id.failUrl))
                            .showError(getString(R.string.invalid_url_format));
                    break;
            }
        }
    }

    @Override
    public void autoFill() {
        ((BoxedEditText)parentView.findViewById(R.id.displayName)).setText("Sample Organization");
        ((BoxedEditText)parentView.findViewById(R.id.externalId)).setText("Sample External ID");
        ((BoxedEditText)parentView.findViewById(R.id.returnUrl)).setText("https://www.google.com.ph");
        ((BoxedEditText)parentView.findViewById(R.id.failUrl)).setText("https://www.hello.com");
    }

    @Override
    public HashMap<String, Object> getFieldsMap() {
        HashMap<String, Object> hashMap = new HashMap<>();
        hashMap.put(DISPLAY_NAME, parentView.findViewById(R.id.displayName));
        hashMap.put(EXTERNAL_ID, parentView.findViewById(R.id.externalId));
        hashMap.put(RETURN_URL, parentView.findViewById(R.id.returnUrl));
        hashMap.put(FAIL_URL, parentView.findViewById(R.id.failUrl));
        return hashMap;
    }

    @Override
    public List<String> getOptionalFields() {
        return new ArrayList(Arrays.asList(DISPLAY_NAME, EXTERNAL_ID, RETURN_URL, FAIL_URL));
    }

    @Override
    public int getPage() {
        return 2;
    }

    public static ClientDetailsFragment newInstance(ScreenListener screenListener) {
        ClientDetailsFragment clientDetailsFragment = new ClientDetailsFragment();
        Bundle bundle = new Bundle();
        bundle.putParcelable(LISTENER, screenListener);
        clientDetailsFragment.setArguments(bundle);
        return clientDetailsFragment;
    }
}

File: app/src/main/java/com/brankas/testapp/fragment/ClientDetailsFragment.kt
package com.brankas.testapp.fragment

import com.brankas.testapp.R
import com.brankas.testapp.`interface`.ScreenListener
import com.brankas.testapp.customview.BoxedEditText

class ClientDetailsFragment: BaseFragment() {
    private lateinit var logoUrl: BoxedEditText
    private lateinit var returnUrl: BoxedEditText
    private lateinit var failUrl: BoxedEditText
    private lateinit var displayName: BoxedEditText

    override fun getLayoutId(): Int {
        return R.layout.fragment_client_details
    }

    override fun initDetails() {
        logoUrl = parentLayout.findViewById(R.id.logoUrl)
        returnUrl = parentLayout.findViewById(R.id.returnUrl)
        failUrl = parentLayout.findViewById(R.id.failUrl)
        displayName = parentLayout.findViewById(R.id.displayName)
    }

    override fun showError(tag: String?) {
        tag?.let {
            when(it) {
                LOGO_URL -> logoUrl.showError(getString(R.string.invalid_url_format))
                RETURN_URL -> returnUrl.showError(getString(R.string.invalid_url_format))
                FAIL_URL -> failUrl.showError(getString(R.string.invalid_url_format))
            }
        }
    }

    override fun autoFill() {
        displayName.setText("Display Name")
        returnUrl.setText("google.com.ph")
        failUrl.setText("www.hello.com")
    }

    override fun getFieldsMap(): HashMap<String, Any> {
        val hashMap = hashMapOf<String, Any>()
        hashMap[DISPLAY_NAME] = displayName
        hashMap[LOGO_URL] = logoUrl
        hashMap[RETURN_URL] = returnUrl
        hashMap[FAIL_URL] = failUrl
        return hashMap
    }

    override fun getOptionalFields(): List<String> {
        return listOf(DISPLAY_NAME, LOGO_URL, RETURN_URL, FAIL_URL)
    }

    override fun getPage(): Int {
        return 2
    }

    companion object {
        const val DISPLAY_NAME = "display_name"
        const val LOGO_URL = "logo_url"
        const val RETURN_URL = "return_url"
        const val FAIL_URL = "fail_url"

        fun newInstance(screenListener: ScreenListener): ClientDetailsFragment {
            return newInstance<ClientDetailsFragment>(screenListener)
        }
    }
}